mono/metadata/class-accessors.c

```diff
@@ -442,6 +446,37 @@ mono_class_has_dim_conflicts (MonoClass *klass)
 	return FALSE;
 }
 
+gboolean
+mono_method_is_constructor (MonoMethod *method)
+{
+	return ((method->flags & CTOR_REQUIRED_FLAGS) == CTOR_REQUIRED_FLAGS &&
+			!(method->flags & CTOR_INVALID_FLAGS) &&
+			!strcmp (".ctor", method->name));
+}
+
+gboolean
+mono_class_has_default_constructor (MonoClass *klass)
+{
```

Comment:
This function should go to class.c, this file is about accessors, which should be fast, and this one isn't.

Reply:
I'll move it.